WalletWasabi.Backend/Controllers/BlockchainController.cs

```diff
@@ -416,4 +416,88 @@ private async Task<StatusResponse> FetchStatusAsync(CancellationToken cancellati
 
 		return status;
 	}
+
+	[HttpGet("get-unconfirmed-transaction-chain")]
+	[ProducesResponseType(200)]
+	[ProducesResponseType(400)]
+	public async Task<List<UnconfirmedTransactionChainItem>> GetUnconfirmedTransactionChainAsync([FromQuery, Required] string transactionId, CancellationToken cancellationToken)
+	{
+		uint256 txId = new(transactionId);
+
+		var cacheKey = $"{nameof(GetUnconfirmedTransactionChainAsync)}_{txId}";
+		var cacheOptions = new MemoryCacheEntryOptions { AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(3) };
+
+		return await Cache.GetCachedResponseAsync(
+			cacheKey,
+			action: (string request, CancellationToken token) => GetUnconfirmedTransactionChainNoChacheAsync(txId, token),
+			options: cacheOptions,
+			cancellationToken);
+	}
+
+	private async Task<List<UnconfirmedTransactionChainItem>> GetUnconfirmedTransactionChainNoChacheAsync(uint256 txId, CancellationToken cancellationToken)
+	{
+		Dictionary<uint256, Transaction> transactionsLocalCache = new();
+		Dictionary<uint256, UnconfirmedTransactionChainItem> unconfirmedTxsChainById = new();
+		var mempool = Global.HostedServices.Get<MempoolMirror>();
+
+		// TODO: Use Transaction cache.
+		var requestedTransaction = await RpcClient.GetRawTransactionAsync(txId, true, cancellationToken);
```

Comment:
As `throwIfNotFound` is `true`,shouldn't a try catch be here?

Reply:
No, I don't think so. 
If we can't get the transaction, we should indicate it to the caller.

Reply:
You should verify if the requested txId is unconfirmed before you advance. It is even cheaper to use MempoolMirror if possible.